opm/material/fluidmatrixinteractions/EclMaterialLawManager.hpp

```diff
@@ -346,8 +358,76 @@ class EclMaterialLawManager
 
             // read the end point scaling info for the saturation region
             unscaledEpsInfo_[satnumIdx].extractUnscaled(deck, eclState, satnumIdx);
+        }
+
+        EclEpsGridProperties epsGridProperties, epsImbGridProperties;
+        epsGridProperties.initFromDeck(deck, eclState, /*imbibition=*/false);
+        if (enableHysteresis())
+            epsImbGridProperties.initFromDeck(deck, eclState, /*imbibition=*/true);
+
+        initParamsForElements_(materialLawParams_, deck, eclState, compressedToCartesianElemIdx, satnumRegionArray,
+                               gasOilUnscaledPointsVector, oilWaterUnscaledPointsVector, gasOilEffectiveParamVector,oilWaterEffectiveParamVector,
+                               epsGridProperties, epsImbGridProperties );
+
 
+        // for well connections
+        std::vector<int> satnumRegionArrayConnections;
+        std::vector<int> compressedToCartesianElemIdxConnections;
+
+        const auto& schedule = eclState.getSchedule();
+        const auto& eclGrid = eclState.getInputGrid();
+
+        size_t timeStep = 0;
+        auto wells = schedule.getWells(timeStep); //need all wells ...
```

Comment:
Wells are only added - never removed, so `getWells( "last_timestep" )` will give you all the wells, that is the same as no-argument overload: `getWells( )`.

That being said your desire to have all the wells makes skeptical - I fear you are assembling an unneccessary data structure - instead of relying on those already found in the parser. There is way to many instances of: "I also want my version of this information" in the opm codebase.